geopandas/geodataframe.py

```diff
@@ -499,7 +506,28 @@ def to_json(self, na="null", show_bbox=False, **kwargs):
           feature individually so that features may have different properties.
         - ``keep``: output the missing entries as NaN.
         """
-        return json.dumps(self._to_geo(na=na, show_bbox=show_bbox), **kwargs)
+        if to_wgs84 is None:
+            if self.crs and not self.crs.equals == 4326:
+                warnings.warn(
+                    "GeoJSON specification requires WGS84 CRS. Geometry will "
+                    "be automatically re-projected in future versions of GeoPandas. "
+                    "To keep existing behavior use 'to_wgs84=False', to silence the "
+                    "warning and get the future behaviour use 'to_wgs_84=True'. "
+                    "GeoDataFrame's CRS:\n"
+                    "{}".format(self.crs.__repr__()),
```

Comment:
We have discussed not to use `__repr__` in warnings. In array crs check we have this instead:

https://github.com/geopandas/geopandas/blob/c4f157d2eb0809653381bfccea5557ed011b05d5/geopandas/array.py#L93-L94